br/pkg/lightning/backend/external/engine.go

```diff
@@ -241,13 +244,16 @@ func (e *Engine) loadIngestData(
 		values = append(values, memBuf.AddBytes(v))
 		cnt++
 		size += len(k) + len(v)
+		totalSize += len(k) + len(v)
 	}
 	if iter.Error() != nil {
 		return nil, errors.Trace(iter.Error())
 	}
 
+	metrics.GlobalSortReadFromCloudStorageRate.WithLabelValues("read_and_sort").Observe(float64(totalSize) / 1024.0 / 1024.0 / time.Since(startTs).Seconds())
```

Comment:
for hot path we should cache the result of `WithLabelValues` to avoid the inner lookup.